packages/scandipwa/src/component/ProductActions/ProductActions.container.js

```diff
@@ -322,15 +322,16 @@ export class ProductActionsContainer extends PureComponent {
         } = customPrice;
 
         const selectedOptions = this.getSelectedOptions();
-        const prices = options.reduce((acc, { data }) => {
-            data.forEach(({ option_type_id, price }) => {
+        const prices = options
+            .flatMap(({ data }) => data)
+            .reduce((acc, option) => {
+                const { option_type_id, price } = option;
                 if (selectedOptions.includes(option_type_id)) {
-                    acc.push(price);
+                    return [...acc, price];
```

Comment:
In this specific case, it might be a bit better to use `acc.push` instead of creating a new array every time.
Data immutability does not matter in this specific point, because the value you generate is completely yours here.
Push will be faster, no additional arrays will be generated.